portal/static/js/utility.js

```diff
@@ -200,46 +180,46 @@ function LRKeyEvent() {
             }
         });
     }
-}
-function appendLREditContainer(target, url, show) { /*global i18next */
-    if (!hasValue(url)) { return false; }
+};
+UtilityObj.prototype.appendLREditContainer = function(target, url, show) { /*global i18next */
+    if (!this.hasValue(url)) { return false; }
     if (!target) { target = $(document); }
     target.append("<div>" +
         "<button class='btn btn-default button--LR'><a href='" + url + "' target='_blank'>" + i18next.t("Edit in Liferay") + "</a></button>" +
         "</div>"
     );
     if (show) { $(".button--LR").addClass("data-show");}
-}
-function __getLoaderHTML(message) {
-    return '<div class="loading-message-indicator"><i class="fa fa-spinner fa-spin fa-2x"></i>' + (hasValue(message) ? "&nbsp;" + message : "") + '</div>';
-}
-function __convertToNumericField(field) {
-    if (field) {
-        if (_isTouchDevice()) {
-            field.each(function() {
-                $(this).prop("type", "tel");
-            });
-        }
+};
+UtilityObj.prototype.getLoaderHTML = function(message) {
+    return "<div class=\"loading-message-indicator\"><i class=\"fa fa-spinner fa-spin fa-2x\"></i>" + (message ? "&nbsp;" + message : "") + "</div>";
+};
+UtilityObj.prototype.convertToNumericField = function(field) {
+    if (!field) {
+        return;
+    }
+    if (this.isTouchDevice()) {
```

Comment:
Would this property be harmful to non-touch devices?

Reply:
thank you ivan, yeah redundant check, as, if the browser support is non-existent, it will fallback to the type of text.